nuclear/cmake/Modules/ToolchainLibraryFinder.cmake

```diff
@@ -13,12 +13,44 @@ function(ToolchainLibraryFinder)
       "VERSION_FILE"
       "VERSION_BINARY_ARGUMENTS"
       "VERSION_REGEX"
+      "LINK_TYPE"
   )
   cmake_parse_arguments(PACKAGE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
   # Clear our required_vars variable
   unset(required_vars)
 
+  if(PACKAGE_LIBRARY OR PACKAGE_LIBRARIES)
+    if(PACKAGE_LINK_TYPE)
+      set(${PACKAGE_NAME}_LINK_TYPE
+          ${PACKAGE_LINK_TYPE}
+          CACHE STRING "Choose method to link the library"
+      )
+    else()
+      set(${PACKAGE_NAME}_LINK_TYPE
+          UNKNOWN
+          CACHE STRING "Choose method to link the library"
+      )
+    endif()
+    set_property(CACHE ${PACKAGE_NAME}_LINK_TYPE PROPERTY STRINGS "SHARED" "STATIC" "MODULE" "UNKNOWN")
```

Comment:
How does this work with the "UNKNOWN" type?

Reply:
"UNKNOWN" will use the default, i.e. search for a `.so` and if it cannot find one search for a `.a`.